provider/sipgate/sipgate.go

```diff
@@ -0,0 +1,77 @@
+package sipgate
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/messagebird/sachet"
+)
+
+var sipgateHTTPClient = &http.Client{Timeout: time.Second * 20}
+
+// Config is the configuration struct for Sipgate provider
+type Config struct {
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+	URL      string `yaml:"url"`
+}
+
+// Sipgate contains the necessary values for the Sipgate provider
+type Sipgate struct {
+	Config
+}
+
+// NewSipgate creates and returns a new Sipgate struct
+func NewSipgate(config Config) *Sipgate {
+	if config.URL == "" {
+		config.URL = "https://api.sipgate.com/v2/sessions/sms"
+	}
+	return &Sipgate{config}
+}
+
+type payload struct {
+	SmsID     string `json:"smsId"`
+	Recipient string `json:"recipient"`
+	Message   string `json:"message"`
+}
+
+// Send sends SMS to user registered in configuration
+func (c *Sipgate) Send(message sachet.Message) error {
+	for _, recipient := range message.To {
+		var body bytes.Buffer
+
+		params := payload{
+			SmsID:     message.From,
+			Recipient: recipient,
+			Message:   message.Text,
+		}
+
+		enc := json.NewEncoder(&body)
+		if err := enc.Encode(params); err != nil {
```

Comment:
Any specific reason for using `json.Encoder.Encode` over `json.Marshal`? I think it'd be good to consider the latter, as this makes it consistent with [other providers](https://github.com/messagebird/sachet/blob/master/provider/freemobile/freemobile.go#L49).
As we're writing the JSON to a buffer, and not a stream, I don't think there's a benefit to this approach..? It all ends up in memory, no matter what approach we take here.

Reply:
I opted to do it that way because `json.NewEncoder` needs an `io.Writer`, and `http.NewRequest` needs an `io.Reader` - so a `bytes.Buffer` is IMHO a cleaner solution than having to wrap a byte slice with `bytes.NewBuffer()`.